lib/mongoid/shardable.rb

```diff
@@ -53,12 +53,10 @@ def shard_key_fields
     # @return [ Hash ] The shard key selector.
     #
     # @api private
-    def shard_key_selector
-      selector = {}
-      shard_key_fields.each do |field|
-        selector[field.to_s] = send(field)
+    def shard_key_selector(prefer_persisted: false)
```

Comment:
I suggest updating the documentation for this method, I think currently id does not describe correctly what the method is doing.